lib/travis/api/v3/remote_query.rb

```diff
@@ -56,7 +56,16 @@ def config
     end
 
     def s3_config
-      raise NotImplemented
+      conf = config.log_options.try(:s3) || {}
```

Comment:
This update doesn't change the code. Unless you were getting the NotImplimented error? The methods defined on the subclass override anything here, that's how caches works with a different config (and why this update didn't break the caches tests). The RemoteQuery base class is config agnostic and the subclass is required to implement it's own config (that's why the error raise is there, for implementations to not rely on the base class config). What errors where you seeing? Were you testing in console and using RemoteQuery instead of Queries::Log? because RemoteQuery isn't supposed to know what config to use, the subclass does.